Shared/Samples/Show viewshed from point on map/ShowViewshedFromPointOnMapView.swift

```diff
@@ -0,0 +1,197 @@
+            let pointSymbol = SimpleMarkerSymbol(style: .circle, color: .red, size: 10)
+            // Sets the renderer to draw the dot symbol on the graphics overlay.
+            let renderer = SimpleRenderer(symbol: pointSymbol)
+            inputGraphicsOverlay.renderer = renderer
+            return inputGraphicsOverlay
+        }()
+        
+        /// Handles the execution of the geoprocessing task and gets the result.
+        var geoprocessingJob: GeoprocessingJob?
+        
+        /// Removes previously tapped location from overlay and draws new dot on tap location.
+        /// - Parameter point: Location that the user tapped on the map.
+        func addGraphic(at point: Point) {
+            inputGraphicsOverlay.removeAllGraphics()
+            let graphic = Graphic(geometry: point)
+            inputGraphicsOverlay.addGraphic(graphic)
+        }
+        
+        /// If the feature set is returned from the geoprocessing, it iterates through each feature and adds it to the
+        /// graphic overlay to display to the user.
+        /// - Parameter resultFeatures: Passes on the results of the geoprocessing so that they can be displayed.
+        func processFeatures(resultFeatures: GeoprocessingFeatures) {
+            if let featureSet = resultFeatures.features {
+                // Iterates through the feature set.
+                for feature in featureSet.features().makeIterator() {
+                    // Creates the graphic for each feature's geometry.
+                    let graphic = Graphic(geometry: feature.geometry)
+                    // Sets the graphic on the overlay to display to the user.
+                    resultGraphicsOverlay.addGraphic(graphic)
+                }
+            }
+        }
+        
+        /// Contains the logic for the geoprocessing and passes the result on to another function to display.
+        /// - Parameter featureCollectionTable: Holds the tapped location feature
+        func performGeoprocessing(_ featureCollectionTable: FeatureCollectionTable) async {
+            let params = GeoprocessingParameters(executionType: .synchronousExecute)
+            // Sets the parameters spatial reference to the point tapped on the map.
+            params.processSpatialReference = featureCollectionTable.spatialReference
+            params.outputSpatialReference = featureCollectionTable.spatialReference
+            // Create a feature with the feature collection table which has the reference to the tapped location.
+            let geoprocessingFeature = GeoprocessingFeatures(features: featureCollectionTable)
+            // Sets the observation point to the tapped location.
+            params.setInputValue(geoprocessingFeature, forKey: "Input_Observation_Point")
+            // Creates geoprocessing job and kicks off process of getting viewshed for the point.
+            geoprocessingJob = geoprocessingTask.makeJob(parameters: params)
+            geoprocessingJob?.start()
```

Comment:
Don't forget to release the reference the job after it finishes by setting it to `nil`. Otherwise, the memory won't be recycled, and the lifespan of  the object will be longer than needed.

Reply:
should be handled now by the defer